search/reverse_geocoder.cpp

```diff
@@ -77,6 +77,20 @@ string Join(string const & s, Args &&... args)
     return s;
   return s + ", " + tail;
 }
+
+ReverseGeocoder::Building FromFeatureImpl(FeatureType & ft, std::string hn, double distMeters)
+{
+  return { ft.GetID(), distMeters, std::move(hn), feature::GetCenter(ft) };
+}
+
+std::string GetHouseNumber(FeatureType & ft)
+{
+  std::string hn = ft.GetHouseNumber();
+  if (hn.empty() && ftypes::IsAddressInterpolChecker::Instance()(ft))
+    hn = ft.GetRef();
+  return hn;
+}
```

Comment:
А нельзя копирования избежать ненужного? Дальше по коду не везде может быть move, например, если empty, то не нужен.
```suggestion
std::string const & GetHouseNumber(FeatureType & ft)
{
  std::string const & hn = ft.GetHouseNumber();
  if (hn.empty() && ftypes::IsAddressInterpolChecker::Instance()(ft))
    return ft.GetRef();
  return hn;
}
```

Reply:
Это усложняет код, дает потенциальное поле для багов, но не дает ничего. Нету оверхеда в пустой строке.